### PR TITLE
Added TikTok provider

### PR DESCRIPTION
**Pull request by tombanksme**
Added a [TikTok Login Kit](https://developers.tiktok.com/doc/login-kit-web) provider.

### CODE REVIEW

File: src/TikTok/Provider.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace SocialiteProviders\Facebook;
```
Comment (Grasume) [inline src/TikTok/Provider.php]: Update this

File: src/TikTok/Provider.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'TIKTOK';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        "user.info.basic"
```
Comment (atymic) [inline src/TikTok/Provider.php:23]: single quotes please

File: src/TikTok/Provider.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return "https://open-api.tiktok.com/platform/oauth/connect?" . http_build_query([
```
Comment (atymic) [inline src/TikTok/Provider.php:31]: also here.

File: src/TikTok/Provider.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    /**
+     * Get TikTok user by token.
+     * 
+     * @param array $data
+     * @return mixed
+     */
+    protected function getUserByToken($data)
+    {
+        // Note: The TikTok api does not currently have a way to get the user data
+        // with only the access token.
```
Comment (atymic) [inline src/TikTok/Provider.php:96]: Not sure what you mean here?

File: src/TikTok/Provider.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+        $response = $this->getHttpClient()->get(
+            "https://open-api.tiktok.com/oauth/userinfo?" . http_build_query([
+                "open_id" => $data[1],
+                "access_token" => $data[0]
```
Comment (atymic) [inline src/TikTok/Provider.php:101]: Is is possible not to pass this through the query string?

**Review by atymic — CHANGES_REQUESTED** [review]

File: src/TikTok/Provider.php (hunk 4 above)
Comment (tombanksme) [reply src/TikTok/Provider.php:96]: The TikTok API does not have an endpoint where it returns the authorized user based on the access token only. You also have to give it the open-id to get the user information.

Thats why I've had to change $data to be an array where $data[0] is the access token and $data[1] is the user's open_id.

File: src/TikTok/Provider.php (hunk 5 above)
Comment (tombanksme) [reply src/TikTok/Provider.php:101]: I'm not too sure what you mean? If it's about the use of http_build_query I just think it looks nicer, I can remove it and just use string composition instead if you would prefer.

File: src/TikTok/Provider.php [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        if ($this->hasInvalidState()) {
+            throw new InvalidStateException();
+        }
+
+        $response = $this->getAccessTokenResponse($this->getCode());
+
+        $token = Arr::get($response, 'data.access_token');
+
+        $this->user = $this->mapUserToObject(
+            $this->getUserByToken([$token, Arr::get($response, 'data.open_id')])
```
Comment (atymic) [inline src/TikTok/Provider.php:56]: Can we key these array params? Just for the sake of readability as `'open_id'      => $data[1],` would be easier to interpret as `'open_id'      => $data['open-id'],`

 :) Thanks

**Review by atymic — COMMENTED** [review]
Just one minor change then good to go, sorry!

**Review by atymic — APPROVED** [review]

**Pull request merged**